### DISCUSSION TITLE
Hello, how does it compare with iterm2?

### DISCUSSION THREAD

**Discussion by lucasjinreal** [wezterm/wezterm#3017] [General]
Hello, how does it compare with iterm2?

**Comment by wez** [comment]
You may wish to take a look at:
* https://github.com/wez/wezterm/discussions/1769

You can and should compare for yourself the list of features on the iterm2 website with the list of features on the wezterm website.

My perception of iterm2:
* A very mature, excellent terminal emulator
* It has tons of features and functions
* it only runs on macOS
* Scriptable using Apple Script and/or Python (which I personally do not enjoy; for me this is slight negative, but for others this could be a positive factor)
* Configuration is via its built-in gui and stored apple's preferences/plist which is awkward to keep in source control and manage across multiple systems. (again: for me this is a slight negative point, but could be positive for others)

wezterm:
* Works essentially the same way with the same config on all platforms; macOS, Linux/Unix/X11/Wayland and Windows 
* Has built-in mosh-plus-tmux-like remote multiplexer and ssh support
* Configuration is lua, making it a good fit for tracking in source control, and directly scriptable
* A high degree of low level control for folks that want/need it
* Supports more inline image protocols than iterm2 (sixel and kitty, as well as iterm2's own protocol)

**Comment by lucasjinreal** [comment]
@wez  thank u, looks very promising on wezterm, while some feature are very important:

1. split windows conveniently (ubuntu system terminal didnt have it);
2. session retore options
3. conveniently support zsh etc
4. Minimal resources on mem and gpu mem.

Does wezterm have all those feature? Especially the last item, very important

**Reply by wez** [reply]
Try it for yourself and find out!

**Reply by kaddkaka** [reply]
>     4. Minimal resources on mem and gpu mem.
Usage on my machine right now:
```
    PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND
  15298 david     20   0 2221,5m 153,7m  85,8m S   1,7   1,1   0:04.06 wezterm-gui
```

**Reply by lucasjinreal** [reply]
@kaddkaka how about iterm2 on your machine

**Reply by kaddkaka** [reply]
Oh, I'm running Linux. I don't have iterm2. Why not compare by yourself?
